platform/core/src/services/MeasurementService/MeasurementServiceAPI.js

```diff
@@ -0,0 +1,76 @@
+import log from '../../log';
+
+export default class MeasurementServiceAPI {
+  constructor({ service }) {
+    this.measurements = {};
+    this.listeners = {};
+
+    if (service) {
+      service.setServiceImplementation({
+        addOrUpdate: this.addOrUpdate.bind(this),
+        subscribe: this.subscribe.bind(this),
+      });
+    }
+  }
+
+  /**
+   * Adds or update persisted measurements.
+   *
+   * @param {MeasurementSchema} schema { id, annotation }
+   */
+  addOrUpdate({ id, annotation, broadcastEventName }) {
+    const isMeasurementValid = this._checkMeasurementValidity(annotation);
+
+    if (!isMeasurementValid) {
+      log.warn(
+        'Attempting to add or update a null/undefined measurement. Exiting early.'
+      );
+      return;
+    }
+
+    let measurementId = id;
+    if (!measurementId) {
+      measurementId = Math.random()
+        .toString(36)
+        .substr(2, 5);
+
+      log.warn(
+        `Measurement ID not set. Using random string ID: ${measurementId}`
+      );
+    }
```

Comment:
I think we have a `uuidv4()` utility lying around, or something similar for generating string GUIDs

Reply:
Updated.